File: programmers/lvl_1/12906.py
# 같은 숫자는 싫어


def solution(arr):
    answer = []
    for item in arr:
        if len(answer) == 0:
            answer.append(item)
        elif answer[-1] != item:
            answer.append(item)

    return answer


# def solution(arr):
#     answer = []
#     for item in arr:
#         if item not in answer:
#             answer.append(item)
#         elif answer[-1] != item:
#             answer.append(item)

#     return answer


print(solution([1, 1, 3, 3, 0, 1, 1]))
print(solution([4, 4, 4, 3, 3]))

File: working_on/12906.js
function solution(arr) {
  let result = [];
  arr.forEach((element) => {
    if (result.length == 0) {
      result.push(element);
    } else if (result[result.length - 1] != element) {
      result.push(element);
    }
  });

  return result;
}
